README.md

```diff
@@ -11,6 +11,20 @@
 
 > An [ESLint](https://eslint.org/) rule that reports usage of deprecated code.
 
+Reports usage of anything marked with a `@deprecated` JSDoc tag (browser APIs, Node.js APIs, library APIs, etc)
+
+**Deprecated browser APIs**
+
+<img width="620" src="https://github.com/gund/eslint-plugin-deprecation/assets/1935696/361d6546-aa74-4b32-b977-36b6e3f183fa" alt="Screenshot of warning `'event' is deprecated` raised by eslint-plugin-deprecation" /><br />

```

Comment:
We should not use screenshots, instead markdown already supports code snippets, you can annotate even the language used in a snippet, for example TS would be marked as \```ts code here```.

I would suggest removing extra examples as they do not provide much value, from just one example you can easily explain what this rule does.

Also as I mentioned in the issue this explanation docs should come after prerequisites and install section because that is the main purpose of the docs.

Reply:
We can use similar structure for examples by showing incorrect/correct versions of the code, for example like [this rule](https://eslint.org/docs/latest/rules/no-await-in-loop).

Reply:
> We should not use screenshots, instead markdown already supports code snippets, you can annotate even the language used in a snippet, for example TS would be marked as ```ts code here```.

Eh, I think it makes it less tangible like this, but...

> We can use similar structure for examples by showing incorrect/correct versions of the code, for example like this rule.

...Since this is a common pattern in other ESLint plugins, I will review the common patterns and follow that one

Reply:
> I would suggest removing extra examples as they do not provide much value, from just one example you can easily explain what this rule does.

Strong disagree, just like the 3 examples of "what types of code could this be?" in the first paragraph, it's important that there are multiple code examples. Not sure if you saw the confusion on Twitter from multiple users, but it's pretty clear people find it hard to understand what this plugin actually is for.

Reply:
2 examples should be fine - one for incorrect and one for correct cases and this is similar to what most eslint rules do.

Reply:
I will take a look at what other plugins do and what your plugin does specially (it surfaces the deprecation messages, which is a unique feature and should be reflected in the documentation), but I'm 👎 on reducing the number of examples. Actually there should be more than 3.

Reply:
To see how deficient your documentation is, just check out a well-documented plugin like `eslint-plugin-unicorn`:

- main docs page: https://github.com/sindresorhus/eslint-plugin-unicorn
- separate rule documentation page (lots of examples): https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat.md

Reply:
@gund as a followup to this, you may want to consider integrating `eslint-doc-generator` to generate a documentation page for the `deprecated` rule, as demonstrated in this PR:

- https://github.com/eslint-community/eslint-plugin-security/pull/101

Reply:
Thanks for linking this one, it could be a nice addition actually, I'm always in for automating stuff.